install/installer/cluster_dev.py

```diff
@@ -110,26 +118,55 @@ def validate(self, document=None, interactive=True):
             )  # Move cursor to end
 
 
-def ask_user(question_name, non_interactive_value=None):
+@typechecked
+class AWSUserNameValidator(Validator):  # pylint: disable=too-few-public-methods
+    """Validate user input"""
```

Comment:
In Python, the convention is for docstrings to follow PEP257. Read it, it's pretty straight-forward. You can also lint them using `flake8-docstrings`/`pydocstyle` and `darglint`.

This one has the following problems:
1) It's not a complete sentence. The first line is the docstring title and must end with a period.
2) Classes represent objects, not actions. Meaning that it shouldn't be imperative but explanatory. It does not *do* anything, it just contains stuff. When documenting functions/methods, you should start with a verb and use an imperative style because those perform actions. But classes are static things. They contain declarations but don't perform actions by themselves.
So instead "It ~does~ something" you should say "It *is* something".
```suggestion
    """Validator for the AWS username input."""
```

Reply:
Interesting, thanks